hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NetworkTopology.java

```diff
@@ -238,8 +239,15 @@ Node getNode(int leafIndex, String scope, List<String> excludedScopes,
    *
    * @param reader    Node where need the data
    * @param nodes     Available replicas with the requested data
-   * @param activeLen Number of active nodes at the front of the array
+   * @param activeLen        Number of active nodes at the front of the array
+   * @param shuffleOperation The algorithm that will be applied to randomize
+   *                         nodes with equal distances
+   *
+   * @return list of sorted nodes if reader is not null,
+   * or shuffled input nodes otherwise. The size of returned list is limited
+   * by activeLen parameter.
    */
   List<? extends Node> sortByDistanceCost(Node reader,
-      List<? extends Node> nodes, int activeLen);
+      List<? extends Node> nodes, int activeLen,
+      Consumer<List<? extends Node>> shuffleOperation);
```

Comment:
I don't think we need to add `shuffleOperation` it to the interface.  It can be set in the constructor of the implementation (normal `shuffle` for production, parameter for the "for test" constructor):

https://github.com/apache/ozone/blob/79b6bb895f4acde9684864a382dd04c9bbf1bea6/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NetworkTopologyImpl.java#L63-L81

Reply:
Thanks, fixed